src/org/usfirst/frc/team1076/robot/SonarReceiver.java

```diff
@@ -0,0 +1,49 @@
+package org.usfirst.frc.team1076.robot;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class SonarReceiver {
+	DatagramSocket socket;
+	int sonarDistance;
+
+	public SonarReceiver(String ip, int portNumber) throws SocketException {
+		socket = new DatagramSocket(null);
+		InetSocketAddress address = new InetSocketAddress(ip, portNumber);
+		socket.bind(address);
+	}
+
+	public void receive() throws IOException {
+		byte[] buffer = new byte[512];
+		DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+		socket.setSoTimeout(1);
+		try {
+		    socket.receive(packet);
```

Comment:
What if this `receive` falls behind? We need to read all of the available packets.